Directory.Build.targets

```diff
@@ -0,0 +1,25 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+	<Target Name="SignPublishFiles"
+			Condition=" '$(MicroBuild_SigningEnabled)' == 'true' "
+			AfterTargets="Publish">
+		<ItemGroup>
+			<FilesToSign Include="$(PublishDir)$(TargetName)$(TargetExt)">
+				<PublishOnly>true</PublishOnly>
+				<Authenticode>Microsoft400</Authenticode>
+			</FilesToSign>
+		</ItemGroup>
+		<SignFiles Files="@(FilesToSign)"
+				   Type="$(SignType)"
+				   BinariesDirectory="$(OutDir)"
+				   IntermediatesDirectory="$(IntermediateOutputPath)"
+				   Condition=" '%(FilesToSign.PublishOnly)' == 'true' " />
+		<ItemGroup>
+			<FilesToSign Remove="@(FilesToSign)" Condition=" '%(FilesToSign.PublishOnly)' == 'true' " />
+		</ItemGroup>
+	</Target>
+
+	<Target Name="SignThirdPartyFiles" BeforeTargets="PackBuildOutputs" DependsOnTargets="Publish">
```

Comment:
Can we just sign these during build rather than publish? I think our publish should only ever sign the publish output

Reply:
Would you just change the DependsOnTargets?

Reply:
I don't think you need this target at all, you can just add a FilesToSign item for the third party libs. You really only need this additional target if you have multiple projects and want to reshare the implementation of signing third party components